test/test_queue.rb

```diff
@@ -24,4 +24,30 @@ def test_flaky_jobs
 
     assert_failures(["./spec/legit_failure_spec.rb[1:3]"], queue)
   end
+
+  def test_fail_fast
+    build_id = rand_id
+
+    Process.wait(start_worker(
+      build_id: build_id, suite: "failing_suite", extra_args: "--fail-fast 1"))
+
+    queue = RSpecQ::Queue.new(build_id, "foo", REDIS_HOST)
+
+    assert_queue_well_formed(queue)
+
+    assert queue.fail_fast_limit_reached?
```

Comment:
Ditto, we also want to ensure that only N jobs were executed and only N failures where encountered.